control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -199,18 +199,28 @@ func (r *GatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 		err := r.updateGatekeeperResources(ctx, log, &gateway, updates.GatewayClassConfig)
 		if err != nil {
+			if isModifiedError(err) {
+				r.Log.V(1).Info("error updating object, will try to re-reconcile")
+
+				return ctrl.Result{}, nil
```

Comment:
I'm assuming that _most_ of the time a modification failure will wind up with a re-reconcile anyway due to the way we trigger things -- but I think you'll want to do something like `return ctrl.Result{Retry: true}, nil` just in case -- this will ensure re-scheduling without the loud errors.

Reply:
Done